kedro/extras/datasets/tensorflow/tensorflow_model_dataset.py

```diff
@@ -109,6 +109,8 @@ def __init__(
         _fs_args = copy.deepcopy(fs_args) or {}
         _credentials = copy.deepcopy(credentials) or {}
         protocol, path = get_protocol_and_path(filepath, version)
+        if protocol == "file":
+            _fs_args["auto_mkdir"] = True
```

Comment:
this will override the users choice, right?
```suggestion
            _fs_args.setdefault("auto_mkdir", True)
```

Reply:
Yes, it will override but if it is set to False the exception will be raised if the target path doesn't exist. What is the use case for such behavior? In the previous `fsspec` version it was set to True by default. 

Note: it is related more to the versioned dataset because it requires a versioned path which doesn't exist (e.g. path `/tmp/test.csv/2019-01-02T00.00.00.000Z/`).

Reply:
yeah so i think to avoid making this a breaking change, we should do the same thing and "set to True by default" -- not "enforce True"

Reply:
Why is this a breaking change? It was set to True in the previous version.
However my main question: what is the use case to set it to `False`. If there is any then I will update it.

Moreover, if you take a look at the previous implementation of e.g. `CSVLocalDataSet` you can see `save_path.parent.mkdir(parents=True, exist_ok=True)`. So we didn't allow users to not create a target path.

Reply:
Was it? From my understanding, we didn't set `auto_mkdir` at all in the previous version and relied on `fsspec` defaulting it to `True`. But if users specified `fs_args: auto_mkdir: param` in their `catalog`, we passed `param` onwards to `fsspec`. This no longer does that -- which makes it a breaking change, no?

I agree that there's pretty much no use case for setting `auto_mkdir` to `False` within Kedro. Maybe we should scope out the `fsspec` bump to a separate PR, instead of doing it here

Reply:
I believe there is no any use case to `auto_mkdir` to `False` in the scope of Kedro and overriding the user's value (if he/she will ever provide it) will prevent from raising an error.